scripts/dl_github_archive.py

```diff
@@ -262,7 +265,7 @@ def download(self):
                     dir0 = os.path.join(dir_untar.path, tarball_prefix)
                     dir1 = os.path.join(dir_untar.path, self.subdir)
                     # submodules check
-                    if self._has_submodule(dir0):
+                    if self.submodules != ['skip'] and self._has_submodule(dir0):
```

Comment:
Does this work when just changing the submodules to checkout? I would guess this triggers an error where it will use the cached tarball with the old submodules.

The list of submodules checked out probably should then be part of the filename to avoid that.

Reply:
Well, the package hash would change and a re-download would be triggered that way.

I can open a follow-up PR to append e.g. a hash of `SUBMODULES` to `FILE`, if there's a consensus that that's the correct way to handle this, but from a quick look I think `FILE` is also used as the default value for `URL_FILE` in places, so I'd be worried about breaking other things.